src/contracts/core/StrategyManager.sol

```diff
@@ -104,102 +93,84 @@ contract StrategyManager is
         }
-
-        // calculate the digest hash
-        bytes32 digestHash = keccak256(abi.encodePacked("\x19\x01", domainSeparator(), structHash));
-
-        /**
-         * check validity of signature:
-         * 1) if `staker` is an EOA, then `signature` must be a valid ECDSA signature from `staker`,
-         * indicating their intention for this action
-         * 2) if `staker` is a contract, then `signature` will be checked according to EIP-1271
-         */
-        EIP1271SignatureUtils.checkSignature_EIP1271(staker, digestHash, signature);
-
         // deposit the tokens (from the `msg.sender`) and credit the new shares to the `staker`
-        shares = _depositIntoStrategy(staker, strategy, token, amount);
+        depositedShares = _depositIntoStrategy(staker, strategy, token, amount);
     }
 
-    /// @inheritdoc IStrategyManager
-    function removeShares(address staker, IStrategy strategy, uint256 shares) external onlyDelegationManager {
-        _removeShares(staker, strategy, shares);
+    /// @inheritdoc IShareManager
+    function removeDepositShares(
+        address staker,
+        IStrategy strategy,
+        uint256 depositSharesToRemove
+    ) external onlyDelegationManager {
+        _removeDepositShares(staker, strategy, depositSharesToRemove);
     }
 
-    /// @inheritdoc IStrategyManager
+    /// @inheritdoc IShareManager
     function addShares(
         address staker,
-        IERC20 token,
         IStrategy strategy,
+        IERC20 token,
         uint256 shares
-    ) external onlyDelegationManager {
-        _addShares(staker, token, strategy, shares);
+    ) external onlyDelegationManager returns (uint256, uint256) {
+        return _addShares(staker, token, strategy, shares);
     }
 
-    /// @inheritdoc IStrategyManager
+    /// @inheritdoc IShareManager
     function withdrawSharesAsTokens(
-        address recipient,
+        address staker,
         IStrategy strategy,
-        uint256 shares,
-        IERC20 token
+        IERC20 token,
+        uint256 shares
     ) external onlyDelegationManager {
-        strategy.withdraw(recipient, token, shares);
+        strategy.withdraw(staker, token, shares);
     }
 
     /// @inheritdoc IStrategyManager
-    function setThirdPartyTransfersForbidden(IStrategy strategy, bool value) external onlyStrategyWhitelister {
-        _setThirdPartyTransfersForbidden(strategy, value);
-    }
-
-    /// @inheritdoc IStrategyManager
-    function setStrategyWhitelister(address newStrategyWhitelister) external onlyOwner {
+    function setStrategyWhitelister(
+        address newStrategyWhitelister
+    ) external onlyOwner {
         _setStrategyWhitelister(newStrategyWhitelister);
     }
 
     /// @inheritdoc IStrategyManager
     function addStrategiesToDepositWhitelist(
-        IStrategy[] calldata strategiesToWhitelist,
-        bool[] calldata thirdPartyTransfersForbiddenValues
+        IStrategy[] calldata strategiesToWhitelist
```

Comment:
won't changing this function signature break the StrategyFactory (which calls this function)? probably need to upgrade it too then?
I believe the same is true for the `removeStrategiesFromDepositWhitelist` function

Reply:
This has been fixed

Reply:
Good callout, we'll need to upgrade that for sure, unless we want to keep this interface unchanged and have an unused input param